### DISCUSSION TITLE
Dynamically set the `url` parameter in .prisma file based on the app environment

### DISCUSSION THREAD

**Discussion by AugustineUmeagudosi** [prisma/prisma#17205] [Q&A] [answered]
I want to maintain different databases for `production`, `development` and `test` and dynamically set the database_url based on the current app environment

Basically, I want to have something like this
```
datasource db {
  provider = "postgresql"
  url      = env("NODE_ENV) === "development" ? env("DATABASE_dev_URL") : env("NODE_ENV) === "production" ? env("DATABASE_URL") : env("DATABASE_TEST_URL")
}
```

Is there any way I can achieve this?

**Comment by nurul3101** [comment, marked as answer]
Hello @AugustineUmeagudosi 👋 

We have a [guide](https://www.prisma.io/docs/guides/development-environment/environment-variables/using-multiple-env-files) on how you can use multiple env files. Each of these env files could point to one specific environment. Would this approach work for your case?

**Reply by AugustineUmeagudosi** [reply]
My case is more like having different values set on the same env file... but the app will dynamically pick which Database_url to use..

For Example,
If I am running test, the app will set the environment variable `NODE_ENV = "test"` and prisma will automatically use the database specified for test to store all test data...

So I kind of need a dynamic switch

**Reply by nurul3101** [reply]
Can you have a look at this [section](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#programmatically-override-a-datasource-url) of Overriding Datasource URL dynamically, I think this would help you in achieving what you want. You can write conditions in the .ts file to read the environment variable and assign the URL accordingly.

**Reply by AugustineUmeagudosi** [reply]
that works... thanks, man!

**Reply by nurul3101** [reply]
No worries! I am glad to hear that this worked for you! 😄

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
